roxctl/common/k8s_port_forward.go

```diff
@@ -0,0 +1,216 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"sort"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/sync"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"k8s.io/kubectl/pkg/polymorphichelpers"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+var (
+	portForwardOnce     sync.Once
+	forwardEndpoint     string
+	portForwardingError error
+
+	errPortForwarding = errox.ServerError.New("port-forwarding error")
+)
+
+const reasonableTimeout = 10 * time.Second
+
+// setKubernetesDefaults sets default values on the provided client config for
+// accessing the Kubernetes API or returns an error if any of the defaults are
+// impossible or invalid.
+// The code is taken from:
+// k8s.io/kubectl@v0.28.2/pkg/cmd/util/kubectl_match_version.go#L114
+func setKubernetesDefaults(config *rest.Config) {
+	// This is allowing the GetOptions to be serialized.
+	config.GroupVersion = &schema.GroupVersion{Group: "", Version: "v1"}
+
+	if config.APIPath == "" {
+		config.APIPath = "/api"
+	}
+	if config.NegotiatedSerializer == nil {
+		// This codec factory ensures the resources are not converted.
+		// Therefore, resources will not be round-tripped through internal
+		// versions. Defaulting does not happen on the client.
+		config.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	}
+	if len(config.UserAgent) == 0 {
+		config.UserAgent = rest.DefaultKubernetesUserAgent()
+	}
+}
+
+func sortByPhase(pods []*corev1.Pod) sort.Interface {
+	return sort.Reverse(podutils.ActivePods(pods))
+}
+
+func getCentralServiceSelectors(ctx context.Context, coreClient coreclient.ServicesGetter, namespace string) (string, error) {
+	svc, err := coreClient.Services(namespace).
+		Get(ctx, "central", metav1.GetOptions{})
+	if err != nil {
+		return "", errors.WithMessage(err, "failed to get central service")
+	}
+	_, sel, err := polymorphichelpers.SelectorsForObject(svc)
+	if err != nil {
+		return "", errors.WithMessage(err, "failed to get central pod selectors")
+	}
+	return sel.String(), nil
+}
+
+func getCentralPod(ctx context.Context, coreClient coreclient.CoreV1Interface, namespace string) (*corev1.Pod, error) {
+	selectors, err := getCentralServiceSelectors(ctx, coreClient, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	forwardablePod, _, err := polymorphichelpers.GetFirstPod(coreClient,
+		namespace, selectors, reasonableTimeout, sortByPhase)
+	if err != nil {
+		return nil, err //nolint:wrapcheck
```

Comment:
I think we could wrap error here
```suggestion
		return nil, errors.Wrap(err, "could not get pod")
```